loadgen/cmd/otelbench/main.go

```
@@ -123,11 +137,16 @@ func main() {
 		}
 	}
 
+	fetcher, err := newElasticsearchStatsFetcher(elasticsearchTelemetryConfig(Config.Telemetry))
+	if err != nil {
+		fmt.Fprintln(os.Stderr, err)
```

non-blocker: I'm thinking if we should fail fast here instead of going ahead with the benchmarking. I see that it isn't possible with the current structure of the code, as an error is returned if the ES telemetry config isn't valid (no connection attempt is made before benchmarks). What I meant by fail fast is to tell between cases
1. telemetry config is not present (user doesn't want telemetry from ES)
2. telemetry config is invalid (e.g. missing credentials) -> fail fast
3. connection failure (e.g. connection error / auth error) -> fail fast
4. telemetry config is present and works ok

marking this as a non-blocker as it doesn't provide additional value if the user is configuring it properly

---

also thank you for using stderr! Great attention to detail

---

Yes, a good idea, I updated ES fetcher to run a quick ping to fail fast if configs are provided.